openbb_platform/providers/tmx/openbb_tmx/utils/helpers.py

```diff
@@ -0,0 +1,1082 @@
+    "^WVSPY": "NASDAQ OMX Alpha WMT vs. SPY Index",
+    "^XAU": "PHLX Gold/Silver Sector",
+    "^XCM": "PHLX Chemicals Sector",
+    "^XEX": "PHLX Europe Sector",
+    "^XND": "Nasdaq-100 Micro Index",
+}
+
+
+async def response_callback(response, _: Any):
+    """Callback for HTTP Client Response."""
+    content_type = response.headers.get("Content-Type", "")
+    if "application/json" in content_type:
+        return await response.json()
+    if "text" in content_type:
+        return await response.text()
+    return await response.read()
+
+
+async def get_data_from_url(
+    url: str,
+    use_cache: bool = True,
+    backend: Optional[SQLiteBackend] = None,
+    **kwargs: Any,
+) -> Any:
+    """Make an asynchronous HTTP request to a static file."""
+    if use_cache is True:
+        async with CachedSession(cache=backend) as cached_session:
+            try:
+                response = await cached_session.get(url, **kwargs)
+                data = await response_callback(response, None)
+            finally:
+                await cached_session.close()
+    else:
+        data = await amake_request(url, response_callback=response_callback, timeout=20)
+
+    return data
+
+
+async def get_data_from_gql(url: str, headers, data, **kwargs: Any) -> Any:
+    """Make an asynchronous GraphQL request."""
+
+    response = await amake_request(
+        url=url,
+        method="POST",
+        response_callback=response_callback,
+        headers=headers,
+        data=data,
+        timeout=30,
+    )
+
+    return response
+
+
+def replace_values_in_list_of_dicts(data):
+    """Helper function to replace "NA" and "-" with None in a list of dictionaries."""
+    for d in data:
+        for k, v in d.items():
+            if isinstance(v, dict):
+                replace_values_in_list_of_dicts([v])  # Recurse into nested dictionary
+            elif isinstance(v, list):
+                for i in range(len(v)):  # pylint: disable=C0200
+                    if isinstance(v[i], dict):
+                        replace_values_in_list_of_dicts(
+                            [v[i]]
+                        )  # Recurse into nested dictionary in list
+                    elif v[i] in ("NA", "-"):
+                        v[i] = None  # Replace "NA" and "-" with None
+            elif v in ("NA", "-"):
+                d[k] = None  # Replace "NA" and "-" with None
+    return data
+
+
+def check_weekday(date) -> str:
+    """Helper function to check if the input date is a weekday, and if not, returns the next weekday.
+
+    Parameters
+    ----------
+    date: str
+        The date to check in YYYY-MM-DD format.
+
+    Returns
+    -------
+    str
+        Date in YYYY-MM-DD format.  If the date is a weekend, returns the date of the next weekday.
+    """
+
+    if pd.to_datetime(date).weekday() not in range(0, 5):
+        date_ = next_workday(pd.to_datetime(date)).strftime("%Y-%m-%d")
+        date = date_
```

Comment:
You do this often where you add an unnecessary dummy variable with an underscore that makes it harder to understand.  You could just have date = ...